src/main/java/games/strategy/engine/framework/startup/login/ClientLogin.java

```diff
@@ -17,6 +28,7 @@
   public static final String ENGINE_VERSION_PROPERTY = "Engine.Version";
   public static final String JDK_VERSION_PROPERTY = "JDK.Version";
   public static final String PASSWORD_PROPERTY = "Password";
+  public static final String ENCRYPTED_PASSWORD_PROPERTY = "RSA Encrypted Password";
```

Comment:
Is using RSA for P2P authentication overkill?  Both nodes share the same secret key, so there's no need to do any kind of key exchange.  Couldn't you just use a symmetric algorithm with the shared key?  For example:

* Server sends a nonce in the challenge.
* Client derives a `SecretKey` from its password.
* Client encrypts the nonce and places it in the response.
* Server derives a `SecretKey` from its password.
* Server decrypts encrypted nonce from client response.
* Server verifies the decrypted nonce is the same value it sent in the challenge.

All we're really trying to do here is verify that both nodes know the same secret.  There should be no need to actually transmit the password.